### PR TITLE
Publishing an openEO process as a service

### PR DESCRIPTION
**Pull request by Pratichhya**
Refreshed @soxofaan's idea of displaying a UDP usecase. The notebook includes section on creating a concrete datacube, inspecting netCDF downloads, and developing a parameterized version stored as a UDP.

### CODE REVIEW

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 1]
```diff
@@ -0,0 +1,2372 @@
...
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# !pip install jupyter-black jupyter\n",
+    "\n",
+    "import jupyter_black\n",
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:39]: I wouldn't impose jupyter-black on people who want to execute the notebook

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 2]
```diff
@@ -0,0 +1,2372 @@
...
+    "%load_ext jupyter_black\n",
+    "\n",
+    "import warnings\n",
+    "warnings.filterwarnings(\"ignore\")"
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:43]: can you add comment why warnings should be ignored? It looks bad style without explanation

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 3]
```diff
@@ -0,0 +1,2372 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0021d684-bcc4-4e68-af6a-2655994d63c1",
+   "metadata": {},
+   "source": [
+    "# Publishing an openEO process as a service \n",
+    "\n",
+    "In this notebook, we want to show how the openEO algorithm can be shared among users. Here, we make use of an apply_dimension process, avoiding the need for extensive merge_cubes. \n",
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:53]: The main goal is how to create a UDP. 
Sharing among users, which is not a standard feature technically, is at most a secondary goal


Also the sentence that connects "apply_dimension" to "avoiding  extensive merge_cubes" is far from obvious and needs a lot more explanation

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 4]
```diff
@@ -0,0 +1,2372 @@
...
+   "source": [
+    "ds[\"VH\"].isel(t=0).plot(vmin=0, vmax=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "502d495b-0d59-40eb-914c-54e9ed9b2dee",
+   "metadata": {},
+   "source": [
+    "This section presented a straightforward example of retrieving and analyzing a corrected Sentinel 1 collection from the backend within a defined area of interest during a specified time frame."
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:754]:
```suggestion
    "This section presented a straightforward example of retrieving and analyzing a corrected Sentinel 1 data cube from the backend within a defined area of interest during a specified time frame."
```

also what does "corrected" mean in this context?

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 5]
```diff
@@ -0,0 +1,2372 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a32ab4d4-e233-411a-b930-b7de9af11a60",
+   "metadata": {},
+   "source": [
+    "## Collect statistics\n",
+    "\n",
+    "As part of more detailed processing within the openEO platform, we'll gather statistical information for the collection. Here, we demonstrate the use of the apply_dimension process along a time dimension."
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:764]:
```suggestion
    "As part of more detailed processing within the openEO platform, we'll gather temporal statistics, using the `apply_dimension` process and a collection of statistical measures (minimum, maximum, quantiles, ...)."
```

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb [hunk 6]
```diff
@@ -0,0 +1,2372 @@
...
...MA5MKvpJCs17YWlknohX6zoZMdLbbUUXMxy5Q93oY5di+gutFGWmKZW/C1qyjYrUKgbEBSrUbNyiRjsa+9LRqhfFSGO1SKjpBWq/sXWiyKXiTyIqDY40IC0ttJvNDB6u1FBgL3QQDRcHa6Lx9qgpZaHqDxNZxlOTnWUmW1jJJNYxWbbKBJj01F42cPIZXRTgIFeDVqpN3RLwVQCvztFbWMvHBxr71+jyzVYLpyd0yCuMEQkEpiVJso0WfmdGkZXHvvQDM7YAotjWcLWDeq1oaIQTY0UbQHW2khoESlaAw3iSyTaChwpf6kM62iIN5LntHddTWG3r3mDf1ogty+ksKOKWbQwhOScx1/Fj0bX87613+KRE2+mMRJy6DxTp2ZsS4fXM2lUuULXbU9p2tJ0iu4nXfaP9vLW+3+fVbe9GTFvc/I5O3jrhbczfs8I2zx9v42kLlFTpsnIHT5eb4JmziQxLxEhzG1JYx+Kcca6Pdhpj4VjFcUT+zX4rQVYXATQEqaBrFZRtqG91qh2++KRbdx++7/w4098kvogWE1Fs0fgLfcwJx3SB0ySu6LuRxFo8B/e/zr6vxEjtxvmPuC3DU9z50Fd5meZVI/t1empaJ4AmsQljEBqsRjdTyiSMyGVlXFdLuf7yPFJhOuTfGoWUa7Rc/8M1WP6KDw8zWMeoGDmrAHGz85RXi81eMuyCLvSuoSuVCHzk934XUnIZ2j22Ey8bpN+hqREhJI3HDyL8uoExdUmhy9bS3I2ZGFDgq4zJ8nvgFXfcimdoBvS73zLx8k/MoXwfMxilbAQpclqDVZ99iC9j2omuVsf0nSMb8jOsuneyzjw/xk8dONWRh8bpHvdHFZTEcaMqP1rExVzyO2tU72/F1uYfPriT1JfnoEg1NEEx2bgAZfvlLdy7YW3E6zsB8siKCT0WheC9fT+2m8tCoGvzKd9hc9U66bfUJ7Ro7/vfe/jiiuuoLe392m3H3/88dx7772Mj48ThiFf+cpXOHz48C9970tf+hLHHHMM2ey/XW7w60UQrBvSC6rroiZndI/PRkOTAjSbMDapu6NYBlYtwJ5vEPTnCYe6ozCo9o5aOdEWmlW1cqdNN8ohybaVKtOa6jC1bTw6jUWc1b6mLhSOjUintSIPdA2m352kcdwI4Uhvp2tRqytSpKyFZUfIYdG2xlvHsMcX8Lri+EMFjdLu7
```
Comment (soxofaan) [inline python/Sentinel1_Stats/Sentinel1_Stats.ipynb:1497]: I would avoid putting so much focus on "sharing with different uses" and "exposing as a service" . These are a feature that are poorly standardized at the moment (and VITO only). The basic reason of existence of UDP's (code reuse) should come first. In a further section the sharing and service aspects can be discussed, but I would keep that separate from the standardized basics

**Review by soxofaan — COMMENTED** [review]
some initial notes

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb (hunk 2 above)
Comment (Pratichhya) [reply python/Sentinel1_Stats/Sentinel1_Stats.ipynb]: I included this for warning when visualizing, but should removed and retested, my mistake I will update it.

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb (hunk 1 above)
Comment (Pratichhya) [reply python/Sentinel1_Stats/Sentinel1_Stats.ipynb]: removed

File: python/Sentinel1_Stats/Sentinel1_Stats.ipynb (hunk 6 above)
Comment (m-mohr) [reply python/Sentinel1_Stats/Sentinel1_Stats.ipynb]: What is a pUpdatedgraph?

Comment (Pratichhya) [reply python/Sentinel1_Stats/Sentinel1_Stats.ipynb]: @m-mohr thank you for noticing 😅. It seems like it was a mistype while adding a comment. It seems to be fine in the PR/notebook itself.

**Comment by m-mohr** [comment]
I just saw that there's nothing in the notebook yet about the Web Editor use. Shall I add that at the end of the notebook? I've improved the Web Editor so that we don't need to fiddle with the URL anylonger. Or how is this meant to be integrated in the OREO workshop? @Pratichhya @soxofaan @patrick-griffiths

**Comment by jdries** [comment]
@m-mohr  I think integrating web editor in the notebook would be a good idea as it would also show the feature for people that check it out afterwards.

@Pratichhya maybe one aspect we could still add is giving an indication of cost (in terms of credits per km²) in the description. This would perhaps encourage udp writers to do the same.

**Comment by m-mohr** [comment]
Added a Web Editor section. Still a bit bland, so I'll add some graphics later today. But the text could get a first review.

**Comment by Pratichhya** [comment]
@jdries, I included some already existing text on credit usage and also on the service description. I am not sure if this is what you mean. Please let me know if it needs to be updated.

**Comment by m-mohr** [comment]
I'm a bit confused right now. Why are there two notebooks in this PR? @Pratichhya

**Comment by Pratichhya** [comment]
@m-mohr, sorry I am messing up today most of the things. It one is by mistake, I will remove it. Thank you notifying me.

**Review by m-mohr — APPROVED** [review]

**Comment by patrick-griffiths** [comment]
@Pratichhya could you please add the oreo_utils.py to the repository where the NB sits?

**Pull request merged**